src/components/MySwitch/MySwitch.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, MouseEvent, useState, useRef, useEffect } from 'react';
+
+type SwitchColor = "primary" | "secondary" | "warning" | "default";
+
+type SwitchSize = "small" | "medium";
+
+interface IMySwitchProps {
+    /**
+    * customize the color of the switch
+    */
+    color?: SwitchColor,
+    /**
+    * customize the size of the switch
+    */
+    size?: SwitchSize,
+    /**
+    * customize the label of switch
+    */
+    label?: string,
+    /**
+    * used to disable the switch
+    */
+    disabled?: boolean,
+    /**
+    * used to check the button
+    */
+    checked?: boolean
+    /**
+    * event handler for the click event of the switch
+    */
+    onChange?: (event: MouseEvent) => void
+}
+
+const MySwitch: FC<IMySwitchProps> = ({
+    color = "primary",
+    size = "medium",
+    label,
+    disabled = false,
+    checked = false,
+    onChange
+}) => {
+
+    const [isChecked, setIsChecked] = useState(checked);
+    
+    const thumbRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        setIsChecked(checked);
+    }, [checked])
+
+    const handleClick = (e: MouseEvent) => {
+        if (disabled) return;
+        setIsChecked(!isChecked);
+        // Add animation after click
+        if (thumbRef && thumbRef.current) {
+            thumbRef.current.style.animation = !isChecked ? `${color}-${size}-effect 0.3s linear` : `Off-${size}-effect 0.3s linear`;
```

Comment:
The ripple color should be the same as the thumb color and the ripple size is also different based on the thumb size. And the ripple should be behind of thumb. I used the outline to do the animation.